lib/ansible/module_utils/network/common/utils.py

```diff
@@ -260,7 +260,10 @@ def dict_diff(base, comparable):
     if not isinstance(base, dict):
         raise AssertionError("`base` must be of type <dict>")
     if not isinstance(comparable, dict):
-        raise AssertionError("`comparable` must be of type <dict>")
+        if not comparable:
```

Comment:
I think you're testing specifically for `None` here? if so do:
 `if comparable is None`

Reply:
`dict_diff` gets called by many of the `nxos/config/*` files. I agree that ideally the callers should not pass `None` to this method, but this method should be protecting itself regardless.